### PR TITLE
[Bug] Fix scenarios where `--empty` flag pushes `limit 0` into metadata queries

### LINKED ISSUE #1033: [Bug] --empty doesn't work when using get_columns_in_relation
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

The `--empty` flag changes the relation name to be something like this:
`(select * from DATABASE_NAME.SCHEMA_NAME.TABLE_NAME where false limit 0)`
This becomes an issue when the adapter `get_columns_in_relation` function is called as it tries to do something like this, which is invalid snowflake sql:
`describe table (select * from DATABASE_NAME.SCHEMA_NAME.TABLE_NAME where false limit 0)`

This means we get the following sql errors when using functions like `dbt_utils.star`:
`Syntax error: unexpected '('. (line 1)`

### Expected Behavior

anywhere where the `{{ relation }}` is used after a describe statement, this should be handled differently for the --empty flag to not be the subquery, but instead use the default original relation.
Like here:
https://github.com/dbt-labs/dbt-snowflake/blob/main/dbt/include/snowflake/macros/adapters.sql#L30

### Steps To Reproduce

1. dbt-snowflake 1.8
2. running with the `--empty` command
3. on any model which includes the `{{ dbt_utils.star() }}` macro

### Relevant log output

_No response_

### Environment

```markdown
- OS: Linux amd64
- Python: 3.10
- dbt-core: 1.8.0
- dbt-snowflake: 1.8.0
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mikealfare**
resolves #1033

### Problem

We use `{{ relation }}` to render the relation name in metadata queries. However, we implemented the `--empty` flag to add `limit 0` when `{{ relation }}` is used. We wind up with queries like:
```sql
describe table (select * from my_relation limit 0)
```
which is not valid sql.

### Solution

- Use `{{ relation.render() }}` in these scenarios (there are many)
- Add a test case that confirms we can use these macros while using the `--empty` flag

### Implementation

1. Merge this PR demonstrating that these tests work (we`re here)
2. Move the bulk of this PR into `dbt-tests-adapter`
3. Create a final PR that removes most of this and uses the `dbt-tests-adapter` tests

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: tests/functional/adapter/empty/test_empty.py [hunk 1]
```diff
@@ -7,3 +11,41 @@ class TestSnowflakeEmpty(BaseTestEmpty):
...
+            "control",
+            "get_columns_in_relation",
+            "alter_column_type",
+            "alter_relation_comment",
+            "alter_column_comment",
+            "alter_relation_add_remove_columns",
+            "truncate_relation",
+        ],
+    )
+    def test_run(self, project, model):
```
Comment (mikealfare) [inline tests/functional/adapter/empty/test_empty.py:50]: Some of these tests pass even without fixing `{{ relation }}` to be `{{ relation.render() }}`. I think it has something to do with whether the template returns just the sql or whether it does the call. I added the tests in case this changes in the future, and updated the macros to use `{{ relation.render() }}`.

File: tests/functional/adapter/empty/test_empty.py
Comment (mikealfare) [inline tests/functional/adapter/empty/test_empty.py:1]: Most of this will move into `dbt-tests-adapter` and then imported here afterwards.

**Review by McKnight-42 — COMMENTED** [review]
This is looking good to me

**Review by colin-rogers-dbt — APPROVED** [review]

**Comment by pempey** [comment]
@mikealfare When might one expect these change in a release?  Or is there a way to test these changes on our system before a release?

**Comment by pempey** [comment]
I will ask my question to a larger audience: @colin-rogers-dbt @McKnight-42 @jtcohen6  When might one expect these change in a release? Or is there a way to test these changes on our system before a release?
